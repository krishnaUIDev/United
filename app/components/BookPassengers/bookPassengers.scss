@import "../../styles/global/_variables";
@import "../../styles/global/_mixins";

.passengersB {
  position: relative;
  padding-left: 24px;
  @media screen and (min-width: $break-point-d) {
    position: absolute;
    top: 40px;
    left: auto;
    width: 300px;
    height: 166px;
    bottom: 16px;
    right: 8px;
    border-radius: 3px;
    padding-left: 40px;
  }
}

.passengerDropdownClearBtn {
  position: absolute;
  font-size: 14px;
  top: 24px;
  right: 24px;
  @media screen and (min-width: $break-point-d) {
    font-size: 12px;
    color: $premier-blue;
    letter-spacing: 0;
    line-height: 24px;
    right: 32px;
    top: 12px;
  }
}

.passengerDropdownCloseBtn {
  position: absolute;
  top: 24px;
  left: 24px;
  @media screen and (min-width: $break-point-d) {
    left: auto; // reset from mobile left positioning
    right: 12px;
    top: 14px;
  }
}

.divider{
  position:absolute;
  left:50%;
  top:10%;
  bottom:10%;
  border-left:1px solid $black-10;
}

.applyBtnContainer {
  position: absolute;
  right: 0;
  left: 0;
  margin-bottom: 24px;
}

.applyBtn {
  width: calc(100% - 48px);
  height: 52px;
  margin-left: 24px;
  font-size: 14px;
  line-height: 11px;
  background: $link-blue;
  color: white;
}

*:not(input) {
  -ms-user-select: none;
}

.srOnly {
  @include visually-hidden;
  text-transform: none;
  font-size: 0;
  line-height: 0;
}
