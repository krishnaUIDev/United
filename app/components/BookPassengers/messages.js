/*
 * BookPassengers Messages
 *
 * This contains all the text for the BookPassengers component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  bookPassengersLegendAriaLabelMultipleSections: {
    id: 'unitedapp.components.BookPassengers.bookPassengersLegendAriaLabelMultipleSections',
    defaultMessage: 'Select number and types of travelers panel opened, use the up or down arrow to increase or decrease number of travelers, or enter the number of travelers in each passenger type, not to exceed total of 9 passengers.',
  },
  bookPassengersLegendAriaLabelNoMaximum: {
    id: 'unitedapp.components.BookPassengers.bookPassengersLegendAriaLabelNoMaximum',
    defaultMessage: 'Select number and types of travelers panel opened, use the up or down arrow to increase or decrease number of travelers, or enter the number of travelers in each passenger type.',
  },
  closePanelAriaLabel: {
    id: 'unitedapp.components.BookPassengers.closePanelAriaLabel',
    defaultMessage: 'Close panel',
  },
});
