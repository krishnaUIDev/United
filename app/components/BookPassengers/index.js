import React from 'react';
import PropTypes from 'prop-types';

import styles from './bookPassengers.scss';
import BookPassengersMenu from '../BookPassengersMenu';
import closeIcon from './assets/close.svg';
import messages from './messages';

class BookPassengers extends React.Component { // eslint-disable-line
  constructor(props) {
    super(props);
    this.state = { decrementTabIndex: -1 };
  }

  componentDidMount() {
    setTimeout(() => {
      if (this.ariaLegend) { this.ariaLegend.focus(); }
    }, 1);
  }

  onPassengerInputBlur() {
    this.props.onSetActiveField('');
    let blur = false;
    setTimeout(() => {
      if (!(document.activeElement) || !(this.isPassengerBtnActive(document.activeElement.id))) {
        this.props.onPassengerInputClick(false);
        blur = true;
      }
    }, 1);
    return blur;
  }

  onQuantityInputFocus(event) {
    this.props.onSetActiveField('');
    const idToSelect = event.target;
    if (idToSelect) {
      // Need timeout for IE
      setTimeout(() => {
        let useSelect = false;
        // Need to use setSelectionRange() for iOS and select() for Chrome (setSelectionRange does not work on Chrome for type="number")
        try {
          idToSelect.setSelectionRange(0, 1);
        } catch (error) {
          useSelect = true;
        }
        if (useSelect) { idToSelect.select(); }
      }, 1);
    }
    this.setState({ decrementTabIndex: 0 });
    return true;
  }

  onClearPassengerInput(event) {
    this.props.onSetActiveField('');
    if (event) {
      event.preventDefault();
    }
    this.props.onChangePassenger(this.props.reducerDefaultToOne, 1);
    for (const index in this.props.clearReducerArray) { // eslint-disable-line
      const currValue = this.props.clearReducerArray[index];
      this.props.onChangePassenger(currValue, 0);
    }
    this.props.onChangeModel(`${this.props.parentModel}${this.props.modelDefaultToOne}`, 1);
    for (const model in this.props.clearModelArray) { // eslint-disable-line
      this.props.onChangeModel(`${this.props.parentModel}.${this.props.clearModelArray[model]}`, 0);
    }
    this.props.onDisableTravelerButtons('plusBtn', false);
    this.props.onDisableTravelerButtons('minusBtn', false);
    return true;
  }

  onApplyButtonClick() {
    this.props.onPassengerInputClick(false);
    setTimeout(() => {
      this.props.onSetActiveField(this.props.mainActiveField);
    }, 1);
    return true;
  }

  onClosePanel() {
    this.props.onPassengerInputClick(false);
    setTimeout(() => {
      this.props.onSetActiveField(this.props.mainActiveField);
    }, 1);
    return true;
  }

  isPassengerBtnActive(activeId) {
    let isActive = false;
    for (const index in this.props.acceptableFocus) { // eslint-disable-line
      const currValue = this.props.acceptableFocus[index];
      if ((activeId) && activeId.indexOf(currValue) >= 0) {
        isActive = true;
      }
    }
    return isActive;
  }

  render() {
    const isDesktop = screen.width > 786;
    const clearBtnTxt = (isDesktop) ? 'Clear All' : 'Clear';
    const sectionTitleA = (isDesktop) ? this.props.isDesktopTitleArrayA : this.props.isMobileTitleArrayA;
    const sectionDetailA = (isDesktop) ? this.props.isDesktopDetailArrayA : this.props.isMobileDetailArrayA;
    let sectionTitleB;
    let sectionDetailB;
    if (this.props.areMultipleSections) {
      sectionTitleB = (isDesktop) ? this.props.isDesktopTitleArrayB : this.props.isMobileTitleArrayB;
      sectionDetailB = (isDesktop) ? this.props.isDesktopDetailArrayB : this.props.isMobileDetailArrayB;
    }
    let legendTab = 0;
    const active = document.activeElement;
    // tabIndex needs to be set to -1 for IE
    if ((active && active.id === 'dropdownContainer') || (!active)) {
      legendTab = -1;
    }
    const ariaLabel = (this.props.areMultipleSections === true) ? messages.bookPassengersLegendAriaLabelMultipleSections.defaultMessage : messages.bookPassengersLegendAriaLabelNoMaximum.defaultMessage;
    return (
      <div
        className={this.props.containerStyle}
        id="dropdownContainer"
        tabIndex="0"
        onBlur={() => this.onPassengerInputBlur()}
      >
        <div>
          <span
            tabIndex={legendTab}
            className={this.props.ariaSectionStyle}
            ref={(input) => { this.ariaLegend = input; }}
            id={this.props.legendId}
            aria-label={ariaLabel}
          >{ariaLabel}</span>
          <div id="passengersSectionA" className={this.props.sectionAstyle}>
            <BookPassengersMenu
              id="passengersA"
              arrayIndexs={this.props.arrayIndexsA}
              ids={this.props.arrayA}
              fullArray={this.props.fullArray}
              sectionTitle={sectionTitleA}
              sectionDetail={sectionDetailA}
              maxLimit={this.props.sectionMaxLimit}
              modelName=".passengers"
              onQuantityInputFocus={(event) => this.onQuantityInputFocus(event)}
              decrementTabIndex={this.state.decrementTabIndex}
              parentModel={this.props.parentModel}
              passengerModelObject={(this.props.passengerModelObject) ? this.props.passengerModelObject : undefined}
              containsInfants={this.props.containsInfants}
              infantTitleA={this.props.infantTitleA}
              infantTitleB={this.props.infantTitleB}
              modelDefaultToOne={this.props.modelDefaultToOne}
              numberOfTravelersTotal={this.props.numberOfTravelersTotal}
              onDisableTravelerButtons={this.props.onDisableTravelerButtons}
              onChangePassenger={this.props.onChangePassenger}
              onChangeModel={(model, value) => this.props.onChangeModel(model, value)}
              onFocusModel={this.props.onFocusModel}
              showTravelerMenu={this.props.showTravelerMenu}
              passengersToBook={this.props.passengersToBook}
              toDisableBtns={this.props.toDisableBtns}
              minValueIsOne={this.props.minValueIsOne}
              individualMaxLimit={(this.props.individualMaxLimit) ? this.props.individualMaxLimit : ''}
              disableIndividualSections={this.props.disableIndividualSections}
            />
          </div>
          {(screen.width > 944 && this.props.showDivider === 'true') ?
            <div className={styles.divider}></div>
          : ''}
          {(this.props.areMultipleSections === 'true') ?
            <div className={styles.passengersB}>
              <BookPassengersMenu
                id="passengersB"
                arrayIndexs={this.props.arrayIndexsB}
                ids={this.props.arrayB}
                fullArray={this.props.fullArray}
                sectionTitle={sectionTitleB}
                sectionDetail={sectionDetailB}
                maxLimit={this.props.sectionMaxLimit}
                individualMaxLimit={(this.props.individualMaxLimit) ? this.props.individualMaxLimit : ''}
                modelName=".passengers"
                onQuantityInputFocus={(event) => this.onQuantityInputFocus(event)}
                decrementTabIndex={this.state.decrementTabIndex}
                parentModel={this.props.parentModel}
                passengerModelObject={(this.props.passengerModelObject) ? this.props.passengerModelObject : {}}
                containsInfants={this.props.containsInfants}
                modelDefaultToOne={this.props.modelDefaultToOne}
                numberOfTravelersTotal={this.props.numberOfTravelersTotal}
                onDisableTravelerButtons={this.props.onDisableTravelerButtons}
                onChangePassenger={this.props.onChangePassenger}
                onChangeModel={(model, value) => this.props.onChangeModel(model, value)}
                onFocusModel={this.props.onFocusModel}
                showTravelerMenu={this.props.showTravelerMenu}
                passengersToBook={this.props.passengersToBook}
                toDisableBtns={this.props.toDisableBtns}
                minValueIsOne={this.props.minValueIsOne}
                infantTitleA={this.props.infantTitleA}
                infantTitleB={this.props.infantTitleB}
                disableIndividualSections={this.props.disableIndividualSections}
              />
            </div>
          : ''}
          <button
            role="button"
            id="clearPassengers"
            tabIndex="0"
            aria-label="Clear all selections"
            className={styles.passengerDropdownClearBtn}
            onClick={(event) => this.onClearPassengerInput(event)}
          >{clearBtnTxt}</button>
          <div className={styles.applyBtnContainer}>
            {(screen.width < 786) ?
              <button
                role="button"
                onClick={() => this.onApplyButtonClick()}
                className={styles.applyBtn}
                aria-label="Apply"
                id="applyPassengersBtn"
              >Apply</button>
            : ''}
          </div>
        </div>
        <button
          title={messages.closePanelAriaLabel.defaultMessage}
          aria-label={messages.closePanelAriaLabel.defaultMessage}
          id="passengersCloseBtn"
          role="button"
          tabIndex="0"
          className={styles.passengerDropdownCloseBtn}
          onClick={() => this.onClosePanel()}
        >
          <img
            src={closeIcon}
            alt=""
            role="presentation"
            aria-hidden="true"
          />
        </button>
      </div>
    );
  }
}

BookPassengers.propTypes = {
  onPassengerInputClick: PropTypes.func,
  onChangePassenger: PropTypes.func,
  arrayA: PropTypes.array,
  arrayB: PropTypes.array,
  fullArray: PropTypes.array,
  onDisableTravelerButtons: PropTypes.func,
  onSetActiveField: PropTypes.func,
  onChangeModel: PropTypes.func,
  reducerDefaultToOne: PropTypes.string,
  clearReducerArray: PropTypes.array,
  modelDefaultToOne: PropTypes.string,
  parentModel: PropTypes.string,
  clearModelArray: PropTypes.array,
  mainActiveField: PropTypes.string,
  isDesktopTitleArrayA: PropTypes.array,
  isMobileTitleArrayA: PropTypes.array,
  isDesktopTitleArrayB: PropTypes.array,
  isDesktopDetailArrayA: PropTypes.array,
  isMobileDetailArrayA: PropTypes.array,
  isMobileTitleArrayB: PropTypes.array,
  isDesktopDetailArrayB: PropTypes.array,
  isMobileDetailArrayB: PropTypes.array,
  containerStyle: PropTypes.string,
  legendId: PropTypes.string,
  arrayIndexsA: PropTypes.array,
  sectionMaxLimit: PropTypes.number,
  areMultipleSections: PropTypes.string,
  showDivider: PropTypes.string,
  arrayIndexsB: PropTypes.array,
  passengerModelObject: PropTypes.object,
  containsInfants: PropTypes.string,
  infantTitleA: PropTypes.string,
  infantTitleB: PropTypes.string,
  numberOfTravelersTotal: PropTypes.string,
  acceptableFocus: PropTypes.array,
  individualMaxLimit: PropTypes.number,
  ariaSectionStyle: PropTypes.string,
  sectionAstyle: PropTypes.string,
  onFocusModel: PropTypes.func,
  passengersToBook: PropTypes.object,
  showTravelerMenu: PropTypes.bool,
  toDisableBtns: PropTypes.object,
  minValueIsOne: PropTypes.string,
  disableIndividualSections: PropTypes.string,
};

export default BookPassengers;
