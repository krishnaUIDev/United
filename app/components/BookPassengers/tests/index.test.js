import React from 'react';
import { shallow } from 'enzyme';
import { spy } from 'sinon';

import BookPassengers from '../index';

describe('<BookPassengers />', () => {
  const onClick = spy();
  const onSetFieldSpy = spy();
  const changeSpy = spy();
  const renderedComponent = shallow(
    <BookPassengers
      arrayA={['Seniors (65+)', 'Infants (under 2)', 'Infants on lap']}
      arrayB={['Children (15-17)', 'Children (12-14)', 'Children (5-11)', 'Children (2-4)']}
      fullArray={['Seniors (65+)', 'Infants (under 2)', 'Infants on lap', 'Children (15-17)', 'Children (12-14)', 'Children (5-11)', 'Children (2-4)']}
      onClick={onClick}
      onPassengerInputClick={onClick}
      onChangePassenger={onClick}
      onDisableTravelerButtons={onClick}
      onSetActiveField={onSetFieldSpy}
      onChangeModel={changeSpy}
      acceptableFocus={['Adults', 'Seniors', 'Infants', 'Children', 'clearPassengers', 'passengersCloseBtn', 'passengersCloseBtn', 'dropdownContainer', 'applyPassengersBtn', 'bookPassengersLegend']}
    />
  );
  it('render BookPassengers Component', () => {
    expect(renderedComponent).toMatchSnapshot();
  });
  it('should render divs', () => {
    expect(renderedComponent.find('div').length).toEqual(4);
  });
  it('should render buttons', () => {
    expect(renderedComponent.find('button').length).toEqual(3);
  });
  it('should render img', () => {
    expect(renderedComponent.find('img').length).toEqual(1);
  });
  it('should handle clear button click', () => {
    const btn = renderedComponent.find('#clearPassengers');
    btn.simulate('click');
    expect(onClick.called).toEqual(true);
    expect(changeSpy.called).toEqual(true);
  });
  it('should handle close button click', () => {
    const btn = renderedComponent.find('#passengersCloseBtn');
    btn.simulate('click');
    expect(onClick.called).toEqual(true);
  });
  it('should handle apply button click', () => {
    const btn = renderedComponent.find('#applyPassengersBtn');
    btn.simulate('click');
    expect(onClick.called).toEqual(true);
  });
  it('should handle blur', () => {
    const btn = renderedComponent.find('#dropdownContainer');
    btn.simulate('blur');
    expect(onClick.called).toEqual(true);
  });
  it('should handle onApplyButtonClick()', () => {
    expect(renderedComponent.instance().onApplyButtonClick()).toEqual(true);
  });
  it('should handle isPassengerBtnActive() with no parameter', () => {
    expect(renderedComponent.instance().isPassengerBtnActive()).toEqual(false);
  });
  it('should handle isPassengerBtnActive() with a parameter', () => {
    expect(renderedComponent.instance().isPassengerBtnActive('Seniors')).toEqual(true);
  });
  it('should handle onClosePanel()', () => {
    expect(renderedComponent.instance().onClosePanel()).toEqual(true);
  });
  it('should handle onClearPassengerInput()', () => {
    expect(renderedComponent.instance().onClearPassengerInput()).toEqual(true);
    expect(changeSpy.called).toEqual(true);
  });
  it('should handle onQuantityInputFocus()', () => {
    expect(renderedComponent.instance().onQuantityInputFocus({ target: { foo: 'bar' } })).toEqual(true);
  });
  it('should handle onPassengerInputBlur()', () => {
    expect(renderedComponent.instance().onPassengerInputBlur()).toEqual(false);
  });
});
