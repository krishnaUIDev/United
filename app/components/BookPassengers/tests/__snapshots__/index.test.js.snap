// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookPassengers /> render BookPassengers Component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={undefined}
    id="dropdownContainer"
    onBlur={[Function]}
    tabIndex="0"
>
    <div>
        <span
            aria-label="Select number and types of travelers panel opened, use the up or down arrow to increase or decrease number of travelers, or enter the number of travelers in each passenger type."
            className={undefined}
            id={undefined}
            tabIndex={0}
        >
            Select number and types of travelers panel opened, use the up or down arrow to increase or decrease number of travelers, or enter the number of travelers in each passenger type.
        </span>
        <div
            className={undefined}
            id="passengersSectionA"
        >
            <BookPassengersMenu
                arrayIndexs={undefined}
                containsInfants={undefined}
                decrementTabIndex={-1}
                disableIndividualSections={undefined}
                fullArray={
                    Array [
                        "Seniors (65+)",
                        "Infants (under 2)",
                        "Infants on lap",
                        "Children (15-17)",
                        "Children (12-14)",
                        "Children (5-11)",
                        "Children (2-4)",
                      ]
                }
                id="passengersA"
                ids={
                    Array [
                        "Seniors (65+)",
                        "Infants (under 2)",
                        "Infants on lap",
                      ]
                }
                individualMaxLimit=""
                infantTitleA={undefined}
                infantTitleB={undefined}
                maxLimit={undefined}
                minValueIsOne={undefined}
                modelDefaultToOne={undefined}
                modelName=".passengers"
                numberOfTravelersTotal={undefined}
                onChangeModel={[Function]}
                onChangePassenger={[Function]}
                onDisableTravelerButtons={[Function]}
                onFocusModel={undefined}
                onQuantityInputFocus={[Function]}
                parentModel={undefined}
                passengerModelObject={undefined}
                passengersToBook={undefined}
                sectionDetail={undefined}
                sectionTitle={undefined}
                showTravelerMenu={undefined}
                toDisableBtns={undefined}
            />
        </div>
        
        
        <button
            aria-label="Clear all selections"
            className={undefined}
            id="clearPassengers"
            onClick={[Function]}
            role="button"
            tabIndex="0"
        >
            Clear
        </button>
        <div
            className={undefined}
        >
            <button
                aria-label="Apply"
                className={undefined}
                id="applyPassengersBtn"
                onClick={[Function]}
                role="button"
            >
                Apply
            </button>
        </div>
    </div>
    <button
        aria-label="Close panel"
        className={undefined}
        id="passengersCloseBtn"
        onClick={[Function]}
        role="button"
        tabIndex="0"
        title="Close panel"
    >
        <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="IMAGE_MOCK"
        />
    </button>
</div>,
  "nodes": Array [
    <div
      className={undefined}
      id="dropdownContainer"
      onBlur={[Function]}
      tabIndex="0"
>
      <div>
            <span
                  aria-label="Select number and types of travelers panel opened, use the up or down arrow to increase or decrease number of travelers, or enter the number of travelers in each passenger type."
                  className={undefined}
                  id={undefined}
                  tabIndex={0}
            >
                  Select number and types of travelers panel opened, use the up or down arrow to increase or decrease number of travelers, or enter the number of travelers in each passenger type.
            </span>
            <div
                  className={undefined}
                  id="passengersSectionA"
            >
                  <BookPassengersMenu
                        arrayIndexs={undefined}
                        containsInfants={undefined}
                        decrementTabIndex={-1}
                        disableIndividualSections={undefined}
                        fullArray={
                              Array [
                                    "Seniors (65+)",
                                    "Infants (under 2)",
                                    "Infants on lap",
                                    "Children (15-17)",
                                    "Children (12-14)",
                                    "Children (5-11)",
                                    "Children (2-4)",
                                  ]
                        }
                        id="passengersA"
                        ids={
                              Array [
                                    "Seniors (65+)",
                                    "Infants (under 2)",
                                    "Infants on lap",
                                  ]
                        }
                        individualMaxLimit=""
                        infantTitleA={undefined}
                        infantTitleB={undefined}
                        maxLimit={undefined}
                        minValueIsOne={undefined}
                        modelDefaultToOne={undefined}
                        modelName=".passengers"
                        numberOfTravelersTotal={undefined}
                        onChangeModel={[Function]}
                        onChangePassenger={[Function]}
                        onDisableTravelerButtons={[Function]}
                        onFocusModel={undefined}
                        onQuantityInputFocus={[Function]}
                        parentModel={undefined}
                        passengerModelObject={undefined}
                        passengersToBook={undefined}
                        sectionDetail={undefined}
                        sectionTitle={undefined}
                        showTravelerMenu={undefined}
                        toDisableBtns={undefined}
                  />
            </div>
            
            
            <button
                  aria-label="Clear all selections"
                  className={undefined}
                  id="clearPassengers"
                  onClick={[Function]}
                  role="button"
                  tabIndex="0"
            >
                  Clear
            </button>
            <div
                  className={undefined}
            >
                  <button
                        aria-label="Apply"
                        className={undefined}
                        id="applyPassengersBtn"
                        onClick={[Function]}
                        role="button"
                  >
                        Apply
                  </button>
            </div>
      </div>
      <button
            aria-label="Close panel"
            className={undefined}
            id="passengersCloseBtn"
            onClick={[Function]}
            role="button"
            tabIndex="0"
            title="Close panel"
      >
            <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="IMAGE_MOCK"
            />
      </button>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <BookPassengers
        acceptableFocus={
                Array [
                        "Adults",
                        "Seniors",
                        "Infants",
                        "Children",
                        "clearPassengers",
                        "passengersCloseBtn",
                        "passengersCloseBtn",
                        "dropdownContainer",
                        "applyPassengersBtn",
                        "bookPassengersLegend",
                      ]
        }
        arrayA={
                Array [
                        "Seniors (65+)",
                        "Infants (under 2)",
                        "Infants on lap",
                      ]
        }
        arrayB={
                Array [
                        "Children (15-17)",
                        "Children (12-14)",
                        "Children (5-11)",
                        "Children (2-4)",
                      ]
        }
        fullArray={
                Array [
                        "Seniors (65+)",
                        "Infants (under 2)",
                        "Infants on lap",
                        "Children (15-17)",
                        "Children (12-14)",
                        "Children (5-11)",
                        "Children (2-4)",
                      ]
        }
        onChangeModel={[Function]}
        onChangePassenger={[Function]}
        onClick={[Function]}
        onDisableTravelerButtons={[Function]}
        onPassengerInputClick={[Function]}
        onSetActiveField={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": BookPassengers {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "acceptableFocus": Array [
            "Adults",
            "Seniors",
            "Infants",
            "Children",
            "clearPassengers",
            "passengersCloseBtn",
            "passengersCloseBtn",
            "dropdownContainer",
            "applyPassengersBtn",
            "bookPassengersLegend",
          ],
          "arrayA": Array [
            "Seniors (65+)",
            "Infants (under 2)",
            "Infants on lap",
          ],
          "arrayB": Array [
            "Children (15-17)",
            "Children (12-14)",
            "Children (5-11)",
            "Children (2-4)",
          ],
          "fullArray": Array [
            "Seniors (65+)",
            "Infants (under 2)",
            "Infants on lap",
            "Children (15-17)",
            "Children (12-14)",
            "Children (5-11)",
            "Children (2-4)",
          ],
          "onChangeModel": [Function],
          "onChangePassenger": [Function],
          "onClick": [Function],
          "onDisableTravelerButtons": [Function],
          "onPassengerInputClick": [Function],
          "onSetActiveField": [Function],
        },
        "refs": Object {},
        "state": Object {
          "decrementTabIndex": -1,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={undefined}
          id="dropdownContainer"
          onBlur={[Function]}
          tabIndex="0"
>
          <div>
                    <span
                              aria-label="Select number and types of travelers panel opened, use the up or down arrow to increase or decrease number of travelers, or enter the number of travelers in each passenger type."
                              className={undefined}
                              id={undefined}
                              tabIndex={0}
                    >
                              Select number and types of travelers panel opened, use the up or down arrow to increase or decrease number of travelers, or enter the number of travelers in each passenger type.
                    </span>
                    <div
                              className={undefined}
                              id="passengersSectionA"
                    >
                              <BookPassengersMenu
                                        arrayIndexs={undefined}
                                        containsInfants={undefined}
                                        decrementTabIndex={-1}
                                        disableIndividualSections={undefined}
                                        fullArray={
                                                  Array [
                                                            "Seniors (65+)",
                                                            "Infants (under 2)",
                                                            "Infants on lap",
                                                            "Children (15-17)",
                                                            "Children (12-14)",
                                                            "Children (5-11)",
                                                            "Children (2-4)",
                                                          ]
                                        }
                                        id="passengersA"
                                        ids={
                                                  Array [
                                                            "Seniors (65+)",
                                                            "Infants (under 2)",
                                                            "Infants on lap",
                                                          ]
                                        }
                                        individualMaxLimit=""
                                        infantTitleA={undefined}
                                        infantTitleB={undefined}
                                        maxLimit={undefined}
                                        minValueIsOne={undefined}
                                        modelDefaultToOne={undefined}
                                        modelName=".passengers"
                                        numberOfTravelersTotal={undefined}
                                        onChangeModel={[Function]}
                                        onChangePassenger={[Function]}
                                        onDisableTravelerButtons={[Function]}
                                        onFocusModel={undefined}
                                        onQuantityInputFocus={[Function]}
                                        parentModel={undefined}
                                        passengerModelObject={undefined}
                                        passengersToBook={undefined}
                                        sectionDetail={undefined}
                                        sectionTitle={undefined}
                                        showTravelerMenu={undefined}
                                        toDisableBtns={undefined}
                              />
                    </div>
                    
                    
                    <button
                              aria-label="Clear all selections"
                              className={undefined}
                              id="clearPassengers"
                              onClick={[Function]}
                              role="button"
                              tabIndex="0"
                    >
                              Clear
                    </button>
                    <div
                              className={undefined}
                    >
                              <button
                                        aria-label="Apply"
                                        className={undefined}
                                        id="applyPassengersBtn"
                                        onClick={[Function]}
                                        role="button"
                              >
                                        Apply
                              </button>
                    </div>
          </div>
          <button
                    aria-label="Close panel"
                    className={undefined}
                    id="passengersCloseBtn"
                    onClick={[Function]}
                    role="button"
                    tabIndex="0"
                    title="Close panel"
          >
                    <img
                              alt=""
                              aria-hidden="true"
                              role="presentation"
                              src="IMAGE_MOCK"
                    />
          </button>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className={undefined}
          id="dropdownContainer"
          onBlur={[Function]}
          tabIndex="0"
>
          <div>
                    <span
                              aria-label="Select number and types of travelers panel opened, use the up or down arrow to increase or decrease number of travelers, or enter the number of travelers in each passenger type."
                              className={undefined}
                              id={undefined}
                              tabIndex={0}
                    >
                              Select number and types of travelers panel opened, use the up or down arrow to increase or decrease number of travelers, or enter the number of travelers in each passenger type.
                    </span>
                    <div
                              className={undefined}
                              id="passengersSectionA"
                    >
                              <BookPassengersMenu
                                        arrayIndexs={undefined}
                                        containsInfants={undefined}
                                        decrementTabIndex={-1}
                                        disableIndividualSections={undefined}
                                        fullArray={
                                                  Array [
                                                            "Seniors (65+)",
                                                            "Infants (under 2)",
                                                            "Infants on lap",
                                                            "Children (15-17)",
                                                            "Children (12-14)",
                                                            "Children (5-11)",
                                                            "Children (2-4)",
                                                          ]
                                        }
                                        id="passengersA"
                                        ids={
                                                  Array [
                                                            "Seniors (65+)",
                                                            "Infants (under 2)",
                                                            "Infants on lap",
                                                          ]
                                        }
                                        individualMaxLimit=""
                                        infantTitleA={undefined}
                                        infantTitleB={undefined}
                                        maxLimit={undefined}
                                        minValueIsOne={undefined}
                                        modelDefaultToOne={undefined}
                                        modelName=".passengers"
                                        numberOfTravelersTotal={undefined}
                                        onChangeModel={[Function]}
                                        onChangePassenger={[Function]}
                                        onDisableTravelerButtons={[Function]}
                                        onFocusModel={undefined}
                                        onQuantityInputFocus={[Function]}
                                        parentModel={undefined}
                                        passengerModelObject={undefined}
                                        passengersToBook={undefined}
                                        sectionDetail={undefined}
                                        sectionTitle={undefined}
                                        showTravelerMenu={undefined}
                                        toDisableBtns={undefined}
                              />
                    </div>
                    
                    
                    <button
                              aria-label="Clear all selections"
                              className={undefined}
                              id="clearPassengers"
                              onClick={[Function]}
                              role="button"
                              tabIndex="0"
                    >
                              Clear
                    </button>
                    <div
                              className={undefined}
                    >
                              <button
                                        aria-label="Apply"
                                        className={undefined}
                                        id="applyPassengersBtn"
                                        onClick={[Function]}
                                        role="button"
                              >
                                        Apply
                              </button>
                    </div>
          </div>
          <button
                    aria-label="Close panel"
                    className={undefined}
                    id="passengersCloseBtn"
                    onClick={[Function]}
                    role="button"
                    tabIndex="0"
                    title="Close panel"
          >
                    <img
                              alt=""
                              aria-hidden="true"
                              role="presentation"
                              src="IMAGE_MOCK"
                    />
          </button>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BookPassengers
    acceptableFocus={
        Array [
            "Adults",
            "Seniors",
            "Infants",
            "Children",
            "clearPassengers",
            "passengersCloseBtn",
            "passengersCloseBtn",
            "dropdownContainer",
            "applyPassengersBtn",
            "bookPassengersLegend",
          ]
    }
    arrayA={
        Array [
            "Seniors (65+)",
            "Infants (under 2)",
            "Infants on lap",
          ]
    }
    arrayB={
        Array [
            "Children (15-17)",
            "Children (12-14)",
            "Children (5-11)",
            "Children (2-4)",
          ]
    }
    fullArray={
        Array [
            "Seniors (65+)",
            "Infants (under 2)",
            "Infants on lap",
            "Children (15-17)",
            "Children (12-14)",
            "Children (5-11)",
            "Children (2-4)",
          ]
    }
    onChangeModel={[Function]}
    onChangePassenger={[Function]}
    onClick={[Function]}
    onDisableTravelerButtons={[Function]}
    onPassengerInputClick={[Function]}
    onSetActiveField={[Function]}
/>,
}
`;
